### ISSUE TITLE
GCS Bucket CORS cannot be removed once created

### ISSUE THREAD

**Issue by tSte** [hashicorp/terraform-provider-google#7967]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to the `modular-magician` user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to `hashibot`, a community member has claimed the issue already.

### Terraform Version

Terraform v0.13.5

### Affected Resource(s)

* google_storage_bucket

### Terraform Configuration Files

```tf
terraform {
  required_version = "0.13.5"
  backend gcs { bucket = "my-playground-terraform-state" }

  required_providers {
    google = { source = "hashicorp/google" }
  }
}

provider google {
  version = "3.49.0"

  project = "my-playground"
  region  = "us-central1"
}

resource google_storage_bucket default {

  name     = "my-testing-bucket-with-cors"
  location = "us-central1" 

  storage_class = "STANDARD"
  versioning { enabled = true }

  force_destroy = true
  lifecycle { prevent_destroy = true }

  cors {
      origin          = ["www.foobar.com"]
      method          = ["GET", "POST"]
      response_header = ["Content-Type"]
      max_age_seconds = 3600
  }
}
```

### Expected Behavior

Terraform will remove CORS bucket settings

### Actual Behavior

Terraform does not remove CORS bucket settings

### Steps to Reproduce

1. Create GCS bucket with CORS
```tf
resource google_storage_bucket default {

  name     = "my-testing-bucket-with-cors"
  location = "us-central1" 

  storage_class = "STANDARD"
  versioning { enabled = true }

  force_destroy = true
  lifecycle { prevent_destroy = true }

  cors {
      origin          = ["www.foobar.com"]
      method          = ["GET", "POST"]
      response_header = ["Content-Type"]
      max_age_seconds = 3600
  }
}
```
and run `terraform apply`.
2. Remove CORS configuration from bucket and run `terraform apply`. You should see planned changes as follows:
```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # google_storage_bucket.default will be updated in-place
  ~ resource "google_storage_bucket" "default" {
        bucket_policy_only          = false
        default_event_based_hold    = false
        force_destroy               = true
        id                          = "my-testing-bucket-with-cors"
        labels                      = {}
        location                    = "US-CENTRAL1"
        name                        = "my-testing-bucket-with-cors"
        project                     = "my-playground"
        requester_pays              = false
        self_link                   = "https://www.googleapis.com/storage/v1/b/my-testing-bucket-with-cors"
        storage_class               = "STANDARD"
        uniform_bucket_level_access = false
        url                         = "gs://my-testing-bucket-with-cors"

      - cors {
          - max_age_seconds = 3600 -> null
          - method          = [
              - "GET",
              - "POST",
            ] -> null
          - origin          = [
              - "www.foobar.com",
            ] -> null
          - response_header = [
              - "Content-Type",
            ] -> null
        }

        versioning {
            enabled = true
        }
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

google_storage_bucket.default: Modifying... [id=my-testing-bucket-with-cors]
google_storage_bucket.default: Modifications complete after 1s [id=my-testing-bucket-with-cors]

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.
```
3. Run `terraform apply` again. You should see identical output.

**Comment by edwardmedia** [comment]
@tSte it is fine with me. In my test, I see the `cors` block is removed. Can you post the debug log?

**Comment by Sytten** [comment]
There was an issue fixed in provider release 3.50.0 about that: https://github.com/hashicorp/terraform-provider-google/releases/tag/v3.50.0
Since the example is using 3.49.0 I would guess this is the issue.

**Comment by edwardmedia** [comment]
@tSte yes. there was a fix in v3.50.0 that might be related. Can you confirm changing the version fixes the problem? If not, please post the debug log

**Comment by tSte** [comment]
@Sytten confirmed - 3.50.0 solves this.
